desktop/docs/electron_architecture.md

```diff
@@ -0,0 +1,114 @@
+## Electron Architecture
+
+### Pinentry
+
+Pinentry will be handled separately from the other desktop MVP features, through
+a new `electron-password` application, which:
+
+1. reads JSON instructions on stdin until EOF
+1. spawns a single window
+1. inside that window, renders HTML from a given path (which can include JS)
+1. prints the user input it received as JSON to stdout
+1. exits
+
+Example stdin:
+```js
+{
+  "html_path": "/usr/local/share/keybase/desktop/templates/password.html",
+  "window_title": "Keybase",
+  "prompt_text": "Please enter your passphrase to unlock your private key",
+  "checkbox": true,
+  "checkbox_text": "Store my passphrase",
+}
+```
+
+Example stdout:
+```js
+{
+  "password": "my_s3kr1t",
+  "checkbox_ticked": false
+}
+```
+
+There's nothing Keybase-specific about this application other than the HTML we
+will choose to point it at.  We could publish it as a totally separate NPM
+module.  We will bundle it as a binary with our installer.  The Go service will
+spawn `electron-password` as a child process.
```

Comment:
Its own binary.  Perhaps having our installer put it into $PATH is the right way to go?